pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -521,6 +532,49 @@ func (l LibvirtWrapper) SetupLibvirt() (err error) {
 	return nil
 }
 
+func getLibvirtLogsFilters() (logFilters string, enableDebugLogs bool) {
+
+	_, envVarExists := os.LookupEnv(services.ENV_VAR_LIBVIRT_DEBUG_LOGS)
+	if !envVarExists {
+		return "", false
+	}
+	launcherVerbosityLevelEnvVar, envVarExists := os.LookupEnv(services.ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY)
+	if !envVarExists {
+		return "", false
+	}
+
+	libvirtLogVerbosity, err := strconv.Atoi(launcherVerbosityLevelEnvVar)
+	if err != nil {
+		log.Log.Infof("cannot apply LIBVIRT_DEBUG_LOGS value %s - must be a number", launcherVerbosityLevelEnvVar)
+		return "", false
+	}
+
+	if libvirtLogVerbosity < 5 {
+		libvirtLogVerbosity = 5
+	}
```

Comment:
@vladikr said [here](https://github.com/kubevirt/kubevirt/pull/7000#discussion_r780442434) that libvirt logs are enabled with verbosity level 5 and above, but if I'm reading the [code](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/template.go#L1194) correctly then it's actually from level **6** instead? So either `templateService.renderLaunchManifest()` needs to be fixed or this should be adjusted accordingly.

Reply:
You're absolutely right, thanks for caching that!

I've given a second thought to it after your comment. Basically the code was problematic in the sense that if the threshold for libvirt logs would have changed (let's say from 5 to 8 in the future) then this code would need to adjust the verbosity levels manually.

Instead, now the two are completely separate. This function defines several internal levels of libvirt logs verbosity starting at 1 (e.g. 1, 2, 3, 4..). These levels correspond to the external virt-launcher threshold (`EXT_LOG_VERBOSITY_THRESHOLD`, which is currently 5). This way, `EXT_LOG_VERBOSITY_THRESHOLD` + i corresponds to internal i-1 level.

This way this code is not dependent libvirt verbosity threshold that could change in the future without extra effort.

Reply:
Yeah this is much better :)